Kajabi/pine
# ci: nightly build improvements

# Description

Updates the nightly `icon-export` process to get the icon library in a state to be published to NPM. Things that occur after the `icon:update` process has run, are as follows:

1. Identify the type of release it will be. If items are renamed or deleted, will be a `major` release, and modified or new will be a `minor` release. 
2. Run a dry-run of `npm version` so that we can get the updated package.json. This will contain the latest `version`.
3. Update the changelog file to include the version in the header.
4. Commit and Tag the latest changes

## Type of change

- [X] Continuous Integration

# How Has This Been Tested?

The node script `update-icon-version.js` has been tested locally by running the following steps.

1. run `npx nx run icons:update`
  a. confirm you have changes after this has been completed
2. run `node ./.scripts/update-icon-version.js`
  a. When this completes you should have a commit that reads something like 
  `ci(icons): v1.0.0, created: , modified: , renamed: , deleted`
  b. if you run `git tag -l` you should see a tag listed for the same version e.g `v1.0.0`
3. run this to reset the branch `git reset --hard HEAD~ && git tag -d v1.0.0`. This will reset the ci commit and delete the tag created.


# Checklist:

If not applicable, leave options unchecked.
